pages/book/statements.mdx

```diff
@@ -277,21 +277,50 @@ foreach (key, value in cells) { // or just k, v
 dump(sum); // 1000
 ```
 
-<Callout type="warning" emoji="⚠️">
+It's also possible to iterate over a map in contract storage, and over maps as members of instances of [Struct](/book/structs-and-messages#structs) or [Message](/book/structs-and-messages#messages) types:
 
-  Note, that at the moment `foreach{:tact}` works only with explicitly provided map identifiers. That is, returning a map from a function and trying to iterate over its entries won't work:
+```tact
+import "@stdlib/deploy";
 
-  ```tact
-  foreach (k, v in emptyMap()) {
-  //               ^ this will give a rather cryptic error message:
-  //                 Syntax error: expected ")"
-  }
-  ```
+struct Fizz { oh_my: map<Int, Int> }
+message Buzz { oh_my: map<Int, Int> }
+
+contract Iterated {
+    oh_my: map<Int, Int>;
+
+    receive("call to iterate!") {
+        let oh_my: map<Int, Int> = emptyMap();
+        oh_my.set(0, 42);
+        oh_my.set(1, 27);
 
-  Trying to iterate over a member of a [Struct](/book/structs-and-messages#structs) also won't work, because struct field access is a compound expression and not an identifier:
+        self.oh_my = oh_my; // assigning local map to the storage one
+        let fizz = Fizz{ oh_my }; // field punning
+        let buzz = Buzz{ oh_my }; // field punning
+
+        // Iterating over map in contract storage
+        foreach (key, value in self.oh_my) {
+            // ...
+        }
+
+        // Iterating over map member of a Struct Fizz instance
+        foreach (key, value in fizz.oh_my) {
+            // ...
+        }
+
+        // Iterating over map member of a Message Buzz instance
+        foreach (key, value in buzz.oh_my) {
+            // ...
+        }
+    }
+}
+```
+
+<Callout type="warning" emoji="⚠️">
+
+  Note, that at the moment `foreach{:tact}` works only with explicitly provided map identifiers or compound expressions, such as field access. That is, returning a map from a function and trying to iterate over its entries won't work:
```

Comment:
It does not work with arbitrary field accesses which is `struct-valued-expression.field`, it only works with explicit id-paths, i.e. constructions like `foo.bar.baz`, where `foo`, `bar` and `baz` are identifiers